sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -447,6 +448,23 @@ pi_result getInfo<const char *>(size_t param_value_size, void *param_value,
                       param_value_size_ret, value);
 }
 
+std::string getKernelNames(pi_program program) {
+  std::string source(program->source_,
+                     program->source_ + program->sourceLength_);
+  std::regex entries_pattern(".entry\\s+([^\\([:s:]]*)");
```

Comment:
I'm not an expert, but is it really the only way to find kernel names?

Reply:
The CUDA driver API does not expose any operations for finding the kernel names of a module. The closest is the ability of getting a kernel by name.

In fact I believe the only place we query all kernel names is when checking if a `program` has a kernel, so asking the backend whether a `program` has the kernel would make for a simpler solution to the problem.

Reply:
Suggest adding a comment describing your idea and leaving as is so far.

Reply:
Good idea. I've added a comment detailing this.